### PR TITLE
Move chat components into its own namespace

### PR DESCRIPTION
**Pull request by ahuang11**
Closes https://github.com/holoviz/panel/issues/5625

It's probably a good idea to move these chat components into its own namespace before release to avoid future deprecations and confusion.

I believe that's everything, but I may be forgetting something so let me know if you think of something that needs to be moved too.

### CODE REVIEW

**Comment by ahuang11** [comment]
Oh I need to actually update the imports.

**Comment by MarcSkovMadsen** [comment]
> Now that it's in its own namespace, would it make sense to drop the "Chat" prefix from ChatInterface, ChatFeed, ChatEntry, etc so that it's `pn.chat.Interface`, `pn.chat.Feed`, and `pn.chat.Entry` or is that too generic?

I believe that is too generic.

File: examples/reference/chat/ChatEntry.ipynb [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "import asyncio\n",
     "\n",
```
Comment (MarcSkovMadsen) [inline examples/reference/chat/ChatEntry.ipynb:10]: I think you need to move the Chat components to their own chat components section in the Component Gallery too?

Comment (ahuang11) [reply examples/reference/chat/ChatEntry.ipynb:10]: I'm a bit confused; these Jupyter notebooks are the component gallery items, right? Once the docs are built, it'll create its own section automatically?

File: panel/chat/langchain.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
     AgentFinish = None
```
Comment (MarcSkovMadsen) [inline panel/chat/langchain.py:9]: You also need to update the links to the reference guides in the docstrings. For example https://panel.holoviz.org/reference/widgets/ChatInterface.html

File: examples/reference/chat/ChatEntry.ipynb (hunk 1 above)
Comment (MarcSkovMadsen) [reply examples/reference/chat/ChatEntry.ipynb:10]: Ok. I did not know. I thought they where defined in some TocTree somewhere.

File: panel/chat/langchain.py (hunk 2 above)
Comment (ahuang11) [reply panel/chat/langchain.py:9]: Thanks for catching that.

File: panel/widgets/chatbox.py [hunk 3]
```diff
@@ -207,7 +207,7 @@ class ChatBox(CompositeWidget):
     The ChatBox widget displays a conversation between multiple users
     composed of users' icons, names, messages, and likes.
 
-    Reference: https://panel.holoviz.org/reference/widgets/ChatBox.html
+    Reference: https://panel.holoviz.org/reference/chat/ChatBox.html
```
Comment (ahuang11) [inline panel/widgets/chatbox.py:210]:
```suggestion
    Reference: https://panel.holoviz.org/reference/widgets/ChatBox.html
```

**Comment by ahuang11** [comment]
Decided to separate each component into its own module since they're all pretty big.

File: examples/reference/chat/ChatEntry.ipynb (hunk 1 above)
Comment (ahuang11) [reply examples/reference/chat/ChatEntry.ipynb:10]: I'm not 100% sure either; leaving this unresolved for Philipp to see.

**Review by MarcSkovMadsen — APPROVED** [review]

File: examples/reference/chat/ChatEntry.ipynb (hunk 1 above)
Comment (philippjfr) [reply examples/reference/chat/ChatEntry.ipynb:10]: You need to add a new section in the `nbsite_gallery_conf` definition of `doc/conf.py`

Comment (ahuang11) [reply examples/reference/chat/ChatEntry.ipynb:10]: Okay added

**Pull request merged**